support-services/src/main/scala/com/gu/support/redemption/corporate/CorporateCodeValidator.scala

```diff
@@ -33,15 +29,15 @@ object GetCodeStatus {
       }
     } yield corporateId
 
-  def withDynamoLookup(dynamoLookup: DynamoLookup): GetCodeStatus = new GetCodeStatus(dynamoLookup)
+  def withDynamoLookup(dynamoLookup: DynamoLookup): CorporateCodeValidator = new CorporateCodeValidator(dynamoLookup)
 
 }
 
-class GetCodeStatus(dynamoLookup: DynamoLookup) extends WithLogging {
+class CorporateCodeValidator(dynamoLookup: DynamoLookup) extends WithLogging {
 
-  import GetCodeStatus._
+  import CorporateCodeValidator._
 
-  def apply(code: RedemptionCode)(implicit ec: ExecutionContext): Future[Either[RedemptionInvalid, CorporateId]] =
+  def validate(code: RedemptionCode)(implicit ec: ExecutionContext): Future[CodeValidationResult] =
```

Comment:
good to stop this being called "apply" but I think could be more positively named e.g. getStatus, as validate doesn't tell you what happens depending whether it's valid or not

Reply:
It is for consistency with the `GiftCodeValidator` but happy to change both to `getStatus`.